pyart/correct/region_dealias.py

```diff
@@ -445,20 +467,30 @@ def _cost_function(nyq_vector, regions, vels_slice,
     return cost
 
 
-def _gradient(nyq_vector, regions, vels_slice, svels_slice, v_nyq_vel,
-              nfeatures):
+def _gradient(nyq_vector, vels_slice_means, svels_slice_means,
+              v_nyq_vel, nfeatures):
     """ Gradient of cost function for minimization
         in region based algorithm """
     gradient_vector = np.zeros(len(nyq_vector))
     i = 0
     for reg in range(nfeatures):
-        add_value = (np.ma.mean(vels_slice) + nyq_vector[i]*v_nyq_vel -
-                     np.ma.mean(svels_slice))
+        add_value = (vels_slice_means[reg] +
+                     np.round(nyq_vector[i])*v_nyq_vel -
+                     svels_slice_means[reg])
+        if(np.isfinite(add_value)):
+            gradient_vector[i] = 2*add_value*v_nyq_vel
+
+        # Regional continuity
+        vels_without_cur = np.delete(vels_slice_means, reg)
+        diffs = np.square(vels_slice_means[reg]-vels_without_cur)
+        the_min = np.argmin(diffs)
+        add_value2 = (vels_slice_means[reg] +
+                      np.round(nyq_vector[i])*v_nyq_vel -
+                      vels_without_cur[the_min])
+
+        if(np.isfinite(add_value2)):
```

Comment:
I believe there is 3 of these extra parenthesis throughout the code, I would just make it:
if np.isfinite(add_values2)